snakemake/settings/types.py

```diff
@@ -348,19 +348,27 @@ class ConfigSettings(SettingsBase):
     config: Mapping[str, str] = immutables.Map()
     configfiles: Sequence[Path] = tuple()
     config_args: Optional[str] = None
+    replace_workflow: bool = False
 
     def __post_init__(self):
         self.overwrite_config = self._get_overwrite_config()
         self.configfiles = self._get_configfiles()
         self.config_args = self._get_config_args()
 
     def _get_overwrite_config(self):
+        from snakemake.logging import logger
+
         overwrite_config = dict()
         if self.configfiles:
             for f in self.configfiles:
                 update_config(overwrite_config, load_configfile(f))
         if self.config:
             update_config(overwrite_config, self.config)
+        if self.replace_workflow and not overwrite_config:
+            logger.warning(
+                "--replace-workflow-config was used but no config entries are provided via "
+                "command line. This flag will be ignored."
+            )
```

Comment:
Better not warn here. People might want to activate this behavior as their default, and then they would always get a warning if they don't specify a --configfile.